src/Sulu/Component/DocumentManager/DocumentManagerInterface.php

```diff
@@ -158,8 +187,9 @@ public function clear();
      * NOTE: This should not be used generally as it exposes the database structure and breaks abstraction. Use the
      *       domain-aware query builder instead.
      *
-     * @param mixed $query Either a JCR-SQL2 string, or a PHPCR query object
+     * @param string|QueryInterface $query Either a JCR-SQL2 string, or a PHPCR query object
```

Comment:
when does this happen? is it maybe always QueryInterface and the string was just false?

In which cases it is a string and in which cases a QueryInterface?

Reply:
I think I have tried this in our implementation and there are still places that use this as a string, sadly.

Reply:
I did have a look at the existing methods call in Sulu at current case I would not add the QueryInterface here as it indeed not designed to have a `PHPCR\Query\QueryInterface` reference here and is never used in sulu core.

Reply:
Oh I see its here handled, then its fine: https://github.com/sulu/sulu/blob/347f0957917f9ac5e91b96fa4d668abe1e12fb6d/src/Sulu/Component/DocumentManager/Subscriber/Phpcr/QuerySubscriber.php#L64-L65

Reply:
but the `getInnerQuery` should then also be `@return string|QueryInterface` on the `QueryCreateEvent`. @mamazu can you add that.

Reply:
Yeah both types are handled in this thing. That's why I extracted them as arguments like that.